### DISCUSSION TITLE
Delete behaviour

### DISCUSSION THREAD

**Discussion by volatilemolotov** [fluxcd/flux2#4931] [General] [answered]
Hello, 

Im a bit confused and also not able to figure out from the docs on how deletion should work. I found that you need to add prune: true to kustomization in order to have the deployed resources deleted after the kustomization itself is deleted. Now what the questions are:
- If kustomization remains in the git will that cause a redeploy? 
- If we suspend it first by adding suspended: true in git will that cause rest of the resources not get deleted 
Thanks

**Comment by makkes** [comment]
> If kustomization remains in the git will that cause a redeploy?

If you delete a bootstrap Kustomization (one that has no parent Kustomization), then Flux will have no way to re-create it. If you delete a Kustomization that has been created by another Kustomization (or a Helm release), it will be re-created the next time the GitRepository resource is reconciled.

**Reply by volatilemolotov** [reply]
Delete how? flux cli delete or remove from git? 

Like the question is can i use CLI to delete a kustomization but somehow keep the files and kustomization in git. Just disable it for now

**Reply by makkes** [reply, marked as answer]
If you want a resource to be deleted from the cluster but keep it in Git, you have several options, depending on how Flux managed the resource. If you have a `kustomization.yaml` in your Git repo, just remove the resource file from the `resources` field. If your Kustomization's `.spec.prune` field is set to `prune`, the resource will be removed from the cluster but you'll still have it in Git.
